### DISCUSSION TITLE
Does GradCAM in mmaction2 support transformer-based models?

### DISCUSSION THREAD

**Discussion by yehuixie** [open-mmlab/mmaction2#2591] [Q&A]
When I use `tools/visualizations/vis_cam.py` for cnn-based models I can get the expected results.
However, it seems that it doesn't work for transformer-based models.
What do I need to do if I want to use Gradcam for Transformer-based models?

**Comment by cir7** [comment]
Currently, it only supports cnn-based models due to the incompatible feature shape with transformed-based models. You can try to adapt the [GradCAM](https://github.com/open-mmlab/mmaction2/blob/main/mmaction/utils/gradcam_utils.py) to transformer-based models, mainly about `_calculate_localization_map` and `_alpha_blending`.

**Reply by yehuixie** [reply]
Thanks a lot. I've solved it.
